test/insert_many_test.rb

```diff
@@ -51,6 +51,21 @@ def teardown
           on_conflict: { column: %i{title author}, do: :update })
       end
     end
+
+    context "when on_conflict.where is provided" do
+      should "perform the correct action based on a partial index" do
+        Book.insert_many [{id: 1, title: "Out of the Silent Planet", author: "C.S. Lewis", isbn: "1974522598", published_on: Date.new(1938, 4, 1) }]
+        Book.insert_many [{id: 2, title: "Perelandra", author: "C.S. Lewis", isbn: "1974522598", published_on: Date.new(1943, 1, 1) }], on_conflict: { column: :isbn, where: "published_on IS NOT NULL", do: :update }
+        assert_equal ["Perelandra"], Book.where(isbn: "1974522598").pluck(:title)
```

Comment:
I think you need to test that if `published_on` _is_ null, the ON CONFLICT clause never needs to get executed

Reply:
Added 👍